anyway/infographics_utils.py

```diff
@@ -490,12 +492,51 @@ def get_latest_accident_date(table_obj, filters):
     return (df.to_dict(orient="records"))[0].get("max_1")  # pylint: disable=no-member
 
 
-def count_accidents_by_car_type(involved_by_vehicle_type_data):  # Temporary for Frontend
-    return [{'car_type': "רכב פרטי", 'percentage_segment': 78,
-            'percentage_country': 74}, {'car_type': "מסחרי/משאית",
-            'percentage_segment': 39, 'percentage_country': 34},
-            {'car_type': "אופנוע", 'percentage_segment': 28,
-            'percentage_country': 15}]
+def percentage_accidents_by_car_type(involved_by_vehicle_type_data):
+    driver_types = defaultdict(int)
+    total_count = 0
+    for item in involved_by_vehicle_type_data:
+        vehicle_type, count = item["involve_vehicle_type"], int(item["count"])
+        total_count += count
+        if vehicle_type in BE_CONST.CAR_VEHICLE_TYPES:
+            driver_types["רכב פרטי"] += count
+        elif vehicle_type in BE_CONST.LARGE_VEHICLE_TYPES:
+            driver_types["מסחרי/משאית"] += count
+        elif vehicle_type in BE_CONST.MOTORCYCLE_VEHICLE_TYPES:
+            driver_types["אופנוע"] += count
+        elif vehicle_type in BE_CONST.BICYCLE_AND_SMALL_MOTOR_VEHICLE_TYPES:
+            driver_types["אופניים/קורקינט"] += count
+        else:
+            driver_types["אחר"] += count
+
+    output = {}
+    for k, v in driver_types.items():  # Calculate percentage
+        output[k] = int(100 * v / total_count)
+
+    return output
+
+
+@lru_cache(maxsize=64)
+def percentage_accidents_by_car_type_national_data_cache(start_time, end_time):
+    involved_by_vehicle_type_data = get_accidents_stats(
+        table_obj=InvolvedMarkerView,
+        group_by="involve_vehicle_type",
+        count="involve_vehicle_type",
+        start_time=start_time,
+        end_time=end_time,
+    )
```

Comment:
@BarVolunteering perhaps it's better to create an aggregated query on the national data, rather then extract all data and then perform aggregation?